hr_timesheet_employee_required/models/account_analytic_line.py

```diff
@@ -1,20 +1,20 @@
-# Copyright 2018 Brainbean Apps (https://brainbeanapps.com)
-# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+# Copyright 2018-2019 Brainbean Apps (https://brainbeanapps.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
 
-from odoo import models
+from odoo import models, api, _
+from odoo.exceptions import ValidationError
+from odoo.tools import config
 
 
 class AccountAnalyticLine(models.Model):
     _inherit = 'account.analytic.line'
 
-    _sql_constraints = [
-        (
-            'project_employee_req',
-            (
-                'CHECK(('
-                '    project_id IS NOT NULL AND employee_id IS NOT NULL'
-                ') OR project_id IS NULL)'
-            ),
-            'Employee is required for Task Log'
-        ),
-    ]
+    @api.multi
+    @api.constrains('project_id', 'employee_id')
+    def _check_employee_id(self):
+        if config['test_enable'] and not self.env.context.get('force_test'):
```

Comment:
Better to use a more specific context key like `test_hr_timesheet_employee_required`, or it can collapse with other tests putting the same context key.

Reply:
Done